src/GenerateAnalyzerRulesets/Program.cs

```diff
@@ -427,62 +435,81 @@ void createAnalyzerSarifFile()
                     writer.Write("$schema", "http://json.schemastore.org/sarif-1.0.0");
                     writer.Write("version", "1.0.0");
                     writer.WriteArrayStart("runs");
-                    writer.WriteObjectStart(); // run
 
-                    writer.WriteObjectStart("tool");
-                    writer.Write("name", "Microsoft Code Analysis");
-                    //writer.Write("version", toolAssemblyVersion.ToString());
-                    //writer.Write("fileVersion", toolFileVersion);
-                    //writer.Write("semanticVersion", toolAssemblyVersion.ToString(fieldCount: 3));
-                    writer.Write("language", culture.Name);
-                    writer.WriteObjectEnd(); // tool
+                    foreach (var assemblymetadata in rulesMetadata)
+                    {
+                        writer.WriteObjectStart(); // run
 
-                    writer.WriteObjectStart("rules"); // rules
+                        writer.WriteObjectStart("tool");
+                        writer.Write("name", assemblymetadata.Key);
+                        writer.Write("version", assemblymetadata.Value.version.ToString());
+                        writer.Write("fileVersion", FileVersionInfo.GetVersionInfo(assemblymetadata.Value.path)?.FileVersion);
```

Comment:
> FileVersionInfo.GetVersionInfo(assemblymetadata.Value.path)?.FileVersion

Does `Write` handle null argument?

Reply:
Good question! I'll change it to not write that if it's null.